HLTrigger/HLTfilters/plugins/L1TTkMuonFilter.cc

```diff
@@ -145,23 +147,32 @@ L1TTkMuonFilter::MuonQualityCut::MuonQualityCut(const edm::ParameterSet& iConfig
   }
 }
 
-bool L1TTkMuonFilter::MuonQualityCut::operator()(const l1t::TkMuon& muon) const {
-  const auto& qualities = allowedQualities_.find(muon.muonDetector());
-  if (qualities != allowedQualities_.end()) {
-    return std::binary_search(qualities->second.begin(), qualities->second.end(), muon.quality());
-  } else {
-    return true;  //if qualities for that detector is not specified, we return true
+bool L1TTkMuonFilter::MuonQualityCut::operator()(const l1t::TrackerMuon& muon) const {
+  bool passesQuality(false);
```

Comment:
This can be moved right before the loop at [L161](https://github.com/cms-sw/cmssw/pull/38442/files#diff-3277d7de81fc938d040d7d0210f4f21c013e0979f4f0b8e1c5699ac51ce79c28R161)